### DISCUSSION TITLE
How do I get the coordinates of the mouse relative to the program?

### DISCUSSION THREAD

**Discussion by zSynctic** [spyoungtech/ahk#292] [Q&A] [answered]
How do I get the coordinates of the mouse relative to the program?

**Comment by spyoungtech** [comment, marked as answer]
In AutoHotkey, the behavior of [`MouseGetPos`](https://www.autohotkey.com/docs/v2/lib/MouseGetPos.htm) (`.mouse_position` or `.get_mouse_position()` in the `ahk` python wrapper) is controlled by the current [CoordMode](https://www.autohotkey.com/docs/v2/lib/CoordMode.htm) setting. If you do not change the CoordMode, the default behavior is to retrieve the coordinates relative to the currently active window.

So, if your program is the active window, there is nothing else you need to do except just call `.mouse_position` or `.get_mouse_position()`.

In the wrapper, there is two ways to change this behavior:

1. Use `.set_coord_mode` to change the global CoordMode setting: `ahk.set_coord_mode(target='Mouse', relative_to='Screen')`  
OR
2. Pass the `coord_mode` parameter to the `.get_mouse_position` method to set the coord mode just for that one call: `ahk.get_mouse_position(coord_mode='Screen')`

As noted in the documentation linked above, you may choose for coord mode to be relative to the `Screen`, `Window`, or `Client`.

If you want the get the mouse coordinates relative to a window that is _not_ active, you can get the mouse position according to the screen, and get the position of the window and then calculate the relative position:

```py
def get_mouse_position_relative_to(win) -> tuple[int, int]:
    win_pos = win.get_position()
    mouse_x, mouse_y = ahk.get_mouse_position(coord_mode='Screen')
    relative_position = (mouse_x - win_pos.x, mouse_y - win_pos.y)
    return relative_position

```
